src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1303,10 +1303,23 @@ async ValueTask<NpgsqlDataReader> ExecuteReaderAsync(CommandBehavior behavior, b
 
             async Task NonMultiplexingWriteWrapper(NpgsqlConnector connector, bool async, CancellationToken cancellationToken2)
             {
-                BeginSend(connector);
-                await Write(connector, async, cancellationToken2);
-                await connector.Flush(async, cancellationToken2);
-                CleanupSend();
+                try
+                {
+                    BeginSend(connector);
+                    await Write(connector, async, cancellationToken2);
+                    await connector.Flush(async, cancellationToken2);
+                    CleanupSend();
+                }
+                catch (OperationCanceledException) when (cancellationToken2.IsCancellationRequested)
+                {
+                    // User requested the cancellation - ignoring the exception, as it's working as expected
```

Comment:
Ok, now I think I understood.

If the application using Npgsql cancels the cancellation token, we swallow the `OperationCanceledException` so that it doesn't need to be caught in their code - yes?

Isn't that an anti-pattern because they should be prepared to catch an `OperationCanceledException` if they cancel an async command?

Reply:
Otherwise they'd have no idea whether their cancellation succeeded or not - no?

Reply:
They would, but from the reader. 

- `Command.ExecuteReaderAsync(token)`
- Cancelling the token
- Got the `OperationCancelledException` from the `ReadBuffer` (exception from which is bubbled up)
- Closing the connection
- On connection closing, we close the reader, and get a second `OperationCancelledException` (unless we swallow it somewhere)

Reply:
Also, right now you'll never get the `OperationCancelledException` (#1146).

Reply:
Ok, I got it thanks!